### ISSUE TITLE
[BUG] Fossify Contacts and Phone loses connection(?!) and incoming calls don't have caller id

### ISSUE THREAD

**Issue by hetszunyukapanyanyimonyok** [FossifyOrg/Phone#71]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Phone/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Phone/issues) or [closed](https://github.com/FossifyOrg/Phone/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Phone/discussions) or [closed](https://github.com/FossifyOrg/Phone/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Phone/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.0

### Affected Android/Custom ROM version

Android 11

### Affected device model

Oneplus

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Telephone rings
2. Only shows phone number caller's name missing 
I can't reproduce this bug consistently.
Because it happens at random, sometimes it does not happen for a long period of time.(3 weeks)

### Expected behavior

When I receive a call to show phone number and name too.

### Actual behavior

Sometimes when I recieve calls, the fossify phone app only shows the phone numbers and the caller's name are missing.(Despite both data in the contacts app.)

### Additional information

Battery optimization is off for both apps!

There is a weak solution:
Restart both Fossify Contacts and Phone, than it works.

Would be a solution to check each other status and show a notification if one of them lose connection to each other?
Maybe better to try to reconnect itself?
Or needs a special permission form system to resart both themself?

**Comment by naveensingh** [comment]
@hetszunyukapanyanyimonyok 

Does the contact show up properly in the Call history? Is it a private contact?

**Comment by hetszunyukapanyanyimonyok** [comment]
@naveensingh 
It is private contact as all of them are. 
When this bug "appears" names are missing from the call history too. (Only numbers shown and date.)

**Comment by naveensingh** [comment]
@Aga-C were you able to reproduce this?

**Comment by Aga-C** [comment]
@naveensingh No, I don't use private contacts, so I've never experienced such problem. I took off `needs triage` because I remember that in SMT people were raising different problems with private contacts performance, so I thought it may be another instance of such.

**Comment by hetszunyukapanyanyimonyok** [comment]
@Aga-C @naveensingh 
For the sake of clarity I will add a few non private/normal phone number and next time when it happens I will report back to you here.

**Comment by naveensingh** [comment]
Ideally, the apps should at least show a toast if some error has occurred while reading the private contact. I'll check this later.

**Comment by hetszunyukapanyanyimonyok** [comment]
@naveensingh Thanks tell me if I can help.(can't develop...)

**Comment by mezhgano** [comment]
Can confirm, got exact same issue on Android 14

**Comment by hetszunyukapanyanyimonyok** [comment]
@mezhgano Could be a different thing. Did you change the setting in Fossify Contacts:
"Show private contacts to Fossify dialer, Fossify SMS and Fossify Calendar." 
Set to ON ?

**Comment by mezhgano** [comment]
@hetszunyukapanyanyimonyok 
Yes, it is ON. 
I don't remember, isn't it set to ON by default?

**Comment by hetszunyukapanyanyimonyok** [comment]
@mezhgano Maybe not, when I need a reinstall I always reset it to on.

**Comment by Aga-C** [comment]
> I don't remember, isn't it set to ON by default?

Yes, it's enabled by default.

**Comment by mezhgano** [comment]
Today i'm recieve another incoming call from unknow number and don't answer.
Right after calling is end notification pop-up that says - Missed Call and "name".
It was **known** contact, calling me about job, but for some reason caller id was not showing on calling screen.

I don't know what's the reason for such behavior, is it bug, or is it something with my phone.
But for sure this is the most critical bug in app right now, it's just unusable.

**Comment by hetszunyukapanyanyimonyok** [comment]
@mezhgano Did your call history show other called/received numbers name? That is a crucial information!

**Comment by mezhgano** [comment]
Planning to perform some testing this evening with second phone, i will post here if catch conditions when or why it is happening...

**Comment by mezhgano** [comment]
Think i got it...

If number was saved in international format, like: 
- +71234567890
- +7 (123) 456-78-90
- 71234567890

It shows caller id everythere and work as expected (as far as i tested).

If number was saved with internal country code ("8" instead "+7/7"), like:
- 81234567890
- 8 (123) 456-78-90 
- 81234567890

It shows caller id in: 
- history
- outgoing call screen
- missed call notification 

It doesn't show caller id in:
- incoming call screen 
- incoming call notification

P.S. Now i must refactor all numbers in contacts to not missing calls or at least keep watching if bug is completely gone...
I think it easier to export contacts in .vcf and do on a desktop PC rather than phone 😄

**Comment by hetszunyukapanyanyimonyok** [comment]
Interesting!
My original bug loses all contacts's name/occasionally despite +44 format.

**Comment by mezhgano** [comment]
Very strange behavior, i will reformat entire contact book and keep checking every incoming call

**Comment by biopsin** [comment]
Still a issue between contacts v1.4.0 and phone v1.8.0
Some contacts are just not recognized with a ID on incoming call and in the phones history later,
also thumbnail show a + not the first letter.

Edit: Both have number formating enabled and flipping makes no effect. 
Im considering if a contact with multiple phone numbers, is what I myself store often and there is a favorite selected, still only the first number originaly stored with the contact will show the ID.

Edit2: After some more eyeballing it's seems to happen on radom even on contact with one phone number now. In call history the countrycode is tripped away also.

**Comment by chesio** [comment]
> also thumbnail show a + not the first letter.

~~When a number is not recognized as a contact, you may add it to contacts by clicking the + icon. Or at least it should work this way, somehow it does not work on my phone in the moment... :-)~~

When a number is not recognized as a contact, the first letter of the number itself is shown in thumbnail (which most of the time is the + sign from country code). See also #631.

**Comment by Efenstor** [comment]
Did anyone experiencing this problem restored user data from a backup? My brother has this problem, I wonder if it's related to [this issue in Fossify Messages](https://github.com/FossifyOrg/Messages/issues/615). I suspect it tries to get caller id from the database using wrong entry id, or something like that, because the database was re-created from ground up on backup restoration.

**Comment by biopsin** [comment]
I did a complete reinstall of android11 and lineage-18.1 and did a restore of messages with both sms and mms from may 2025 in desember and so far things are in sync for me.
Edit: at some point not long ago it just lost sync on contacts with several numbers registred.
